jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
```

Comment:
That `andThen` example is working on an integer (the return of `head months`) and the function `toValidMonth`, that seems superfluous.  Wouldn't it be easier to just turn that example in to:
```elm
toValidMonth : Int -> Maybe Int
toValidMonth month =
    if month >= 1 && month <= 12 then
        Just month
    else
        Nothing

getFirstMonth : List Int -> Maybe Int
getFirstMonth months =
    head months
    |> toValidMonth
```
Thus I do not see what `andThen` brings there?

_/me is assuming that `head` is implemented like (in ocaml syntax ignoring error checking) `let head (first :: _) = first`_